static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -0,0 +1,157 @@
+import {useCallback, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {ModalRenderProps} from 'sentry/actionCreators/modal';
+import Button, {ButtonPropsWithoutAriaLabel} from 'sentry/components/button';
+import {t} from 'sentry/locale';
+
+// This is just a doubly linked list of steps
+interface OnboardingStep {
+  current: React.ComponentType<OnboardingStepProps>;
+  next: OnboardingStep | null;
+  previous: OnboardingStep | null;
+}
+
+type OnboardingRouterState = [OnboardingStep, (step: OnboardingStep | null) => void];
+function useOnboardingRouter(initialStep: OnboardingStep): OnboardingRouterState {
+  const [state, setState] = useState(initialStep);
+
+  const toStep = useCallback((nextStep: OnboardingStep | null) => {
+    // For ergonomics, else we need to move everything to consts so that typescript can infer non nullable types
+    if (nextStep === null) {
+      return;
+    }
+
+    setState(current => {
+      const next = {...nextStep, next: null, previous: current};
+      // Add the edges between the old and the new step
+      current.next = next;
+      next.previous = current;
+      // Return the neext step
```

Comment:
```suggestion
      // Return the next step
```

Reply:
I fixed this and the followup comments below. I only kept them as placeholders, they should be all t'd up in https://github.com/getsentry/sentry/pull/37118/files and I also changed the content to what we had in the designs.